src/core_modules/capture-core/components/Pages/MainPage/TeiWorkingLists/epics/teiViewEpics/lib/getTeiListData/getTeiListData.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import { convertToClientTeis } from './convertToClientTeis';
+import { getTeisWithSubvalues } from './getTeisWithSubvalues';
+import type { RawQueryArgs, RawFilterQueryArgs } from './types';
+import type { InputMeta } from './getTeiListData.types';
+import type { TeiColumnsMetaForDataFetching } from '../../../../types';
+
+const getApiFilterQueryArgs = (filters?: RawFilterQueryArgs = {}): ?{| filter: Array<string> |} => {
+    const apiFilterQueryArgs =
+            Object
+                .keys(filters)
+                .flatMap((filterKey) => {
+                    const filter = filters[filterKey];
+                    // not handling any main columns yet so this is pretty straight forward
+                    if (Array.isArray(filter)) {
+                        return filter
+                            .map(filterPart => `${filterKey}:${filterPart}`);
+                    }
+                    return `${filterKey}:${filter}`;
+                });
+
+    return (apiFilterQueryArgs.length > 0 ? { filter: apiFilterQueryArgs } : null);
+};
+
+const getApiOrderById = (sortById: string, columnsMetaForDataFetching: TeiColumnsMetaForDataFetching) => {
+    const { id, apiName } = columnsMetaForDataFetching.get(sortById) || {};
+    return apiName || id;
+};
+
+const getApiOrderByQueryArgument = (sortById: string, sortByDirection: string, columnsMetaForDataFetching: TeiColumnsMetaForDataFetching) => {
+    const apiId = getApiOrderById(sortById, columnsMetaForDataFetching);
+    return `${apiId}:${sortByDirection}`;
```

Comment:
You're right if you look at this code in isolation. But the `columnsMetaForDataFetching.get(sortById)` will always return something, I only added the `{}` on line 26 because of flow.